src/Client/ClientBase.cpp

```diff
@@ -647,6 +647,12 @@ void ClientBase::processTextAsSingleQuery(const String & full_query)
 
 void ClientBase::processOrdinaryQuery(const String & query_to_execute, ASTPtr parsed_query)
 {
+    if (fake_drop)
+    {
+        if (parsed_query->as<ASTDropQuery>())
+            return;
```

Comment:
Does it make sense to replace `DROP` with `TRUNCATE` for table engines like `Memory` or `Join` to reduce memory usage?

Reply:
But how we will know which table engine is used by drop query?

Reply:
By executing one more query to get table engine or by moving this logic to server (but it's questionable)

Reply:
Makes sense, I will try it (created [issue](https://github.com/ClickHouse/ClickHouse/pull/27928#discussion_r929901743) to not forget)